scripts/generate_release_updates.py

```diff
@@ -0,0 +1,167 @@
+    '   %s.\n'
+    '   %s\n'
+    '   Finally, I\'d like to thank %s for their help with pre-release '
+    'testing, bug-fixing and QA, as well as %s for leading the QA team for '
+    'this release.\n\n'
+    'Thanks,\n'
+    '%s\n') % tuple(SECTIONS_TO_ADD)
+
+PARENT_DIR = os.path.abspath(os.path.join(os.getcwd(), os.pardir))
+RELEASE_MAIL_MESSAGE_FILEPATH = os.path.join(
+    PARENT_DIR, 'release_mail_message.txt')
+
+
+def create_new_file_with_release_message_template():
+    """Adds the template message to release mail filepath."""
+    with python_utils.open_file(RELEASE_MAIL_MESSAGE_FILEPATH, 'w') as f:
+        f.write(RELEASE_MAIL_MESSAGE_TEMPLATE)
+
+    common.ask_user_to_confirm(
+        'Please make updates to following file %s for generating the '
+        'release announcement mail by adding:\n'
+        '   1. Main changes for release\n'
+        '   2. Editorials/announcements if required\n'
+        '   3. Author details from release_summary.md\n'
+        '   4. Names of release testers\n'
+        '   5. Name of QA Team lead\n'
+        '   6. Your name\n' % RELEASE_MAIL_MESSAGE_FILEPATH)
+
+
+def validate_release_message():
+    """Checks the message after the mail template is updated by the
+    user.
+
+    Raises:
+        Exception: The message still contains sections from template
+            which are not updated.
+    """
+    with python_utils.open_file(RELEASE_MAIL_MESSAGE_FILEPATH, 'r') as f:
+        release_mail_message = f.read()
+        extra_sections = [
+            section for section in SECTIONS_TO_ADD if (
+                section in release_mail_message)]
+        if extra_sections:
+            raise Exception(
+                'Template not formatted correctly. '
+                'Following sections still not updated: %s.\n'
+                'Please re-run the scripts and make the updates again.' % (
+                    ', '.join(extra_sections)))
+
+
+
+def get_new_authors_and_contributors_mail_ids():
+    """Returns the mail ids of new authors and contributors for the release.
+
+    Returns:
+        list(str). List of mail ids of new authors and contributors
+            for the release.
+    """
+    with python_utils.open_file(
+        release_constants.RELEASE_SUMMARY_FILEPATH, 'r') as f:
+        release_summary_lines = f.readlines()
+
+    new_authors_and_contributors_mail_ids = []
+    for line_text in ['### New Authors:\n', '### New Contributors:\n']:
+        start_index = release_summary_lines.index(line_text)
+        end_index = start_index
+        for index, line in enumerate(release_summary_lines[start_index + 1:]):
+            if line.startswith('###'):
+                end_index = end_index + index
+                break
+
+        new_details_list = release_summary_lines[start_index + 1: end_index]
+        new_authors_and_contributors_mail_ids.extend([
+            detail[detail.find('<') + 1: detail.find('>')] for detail in (
```

Comment:
Break before "for", then you don't need the additional paren.

Reply:
Done.